plugins/lookup/bitwarden.py

```diff
@@ -75,6 +94,16 @@
   ansible.builtin.debug:
     msg: >-
       {{ lookup('community.general.bitwarden', 'a_test', field='password', bw_session='bXZ9B5TXi6...') }}
+
+- name: "Get 'password' from all Bitwarden records named 'a_test', create missing records with an autogenerated, 32 character password"
+  ansible.builtin.debug:
+    msg: >-
+      {{ lookup('community.general.bitwarden', 'a_test', field='password', generate_if_missing='--length 32') }}
+
+- name: "Get 'username' from all Bitwarden records named 'a_test', creating missing records with predefined username 'admin'"
+  ansible.builtin.debug:
+    msg: >-
+      {{ lookup('community.general.bitwarden', 'a_test', field='username', set='admin') }}
```

Comment:
This will always be `admin`, right?

Reply:
yes. I agree that a "lookup" plugin setting variables seems weird, but as I mentioned the `passwordstore` lookup already does this, albeit only with generate_if_missing.

The option to set entries makes sense in my mind, as Bitwarden records store more than actual, hopefully autogenerated secrets. You have notes, uris, usernames etc.

For me, it is really nice to be able to for example deploy a wordpress via ansible and set the url, username + password directly in bitwarden and ansible alike, in host_vars:
```
wordpress_sites:
  example.com:
    url: "{{ lookup('community.general.bitwarden', 'Wordpress www.example.com', field='uris', set='www.example.com') }}"
    admin_user: "{{ lookup('community.general.bitwarden', 'Wordpress www.example.com', field='username', set='admin') }}"
    admin_password: "{{ lookup('community.general.bitwarden', 'Wordpress www.example.com', field='password', generate_if_missing='--length 32') }}"
    database_password: "{{ lookup('community.general.bitwarden', 'Wordpress www.example.com', field='database_password', generate_if_missing='--length 32') }}"
```

The fields via `set` will usually not be passwords, in the same usecase as the uri-field will probably not be autogenerated but set.

That means you can have nice Bitwarden entries while only using ansible, for other, more frontend teams to use. Also, as ansible is usually the source of truth on how the infrastructure is deployed, it would make sense that ansible controls the content of its Bitwarden records.
But, it might also be an argument to have a `set_if_missing` instead, which would mean that Bitwarden is the source of truth, not ansible (as with`generate_if_missing`). That might be more true to the "lookup" perspective.

What are your opinions on that?

Reply:
I did use the bitwarden lookup a bit more in my deployments and I am revising my opinion.

I do think that `set` in the way introduced in this PR would yield some benefits, BUT:
This lookup is a wrapper around `bw`, the CLI for Bitwarden, which is _that_ slow that it does not make sense to use it without grouping significant chunks of records together, as any call of the lookup takes about a second, even for generating a local password (or running `bw --version`, for that matter).
If you inventory should not take multiple minutes to read some passwords via Bitwarden, you cannot use `field=`, but group your secrets together (by service, for example), read the entire record and pick through the json with Jinja2. I'm very disappointed...

Therefore, you cannot in praxis reference every individual field via lookup, so `set` becomes useless, as does `generate_if_missing` in most capacities. But as the lookup does primarily filter the results, I'd keep the generate but scrap the `set`, if someone does not have a different opinion?

Reply:
I wrote a [wrapper](https://github.com/lungj/rbw_wrapper) around [`rbw`](https://github.com/doy/rbw) to make the CLI behave like `bw`. However, I switched away from bitwarden so I no longer maintain the wrapper.

Reply:
yeah I totally get the need to use rbw instead...

I really don't see this PR anymore for this module, as it is unusable in praxis. But I do see a need for a `bitwarden_rust` module, as the current bitwarden lookup causes people to waste time finding out it is not feasable and then checking out `rbw` - which we could fix by providing a lookup for `rbw` from the start, mentioning the differences in the docs?

Who should be asked if that'd be a welcomed module, before someone invests time in work?

Reply:
I actually wrote this bitwarden lookup module and was using it in practice in conjunction with the rbw wrapper I linked to above. I'm not a patient person and I had around 100-200 calls to this module with the `rbw` wrapper in most of my playbooks (though I've since rearchitected my systems to cut this number down by a factor of 10).

However, the `rbw-agent` portion of `rbw` on my old daily driver was unstable (but it's been perfectly fine on my other machines), so I threw in the towel and moved back to HCP.

Reply:
Yes I expect this lookup to work fine with the `rbw` wrapper. But it requires a wrapper workaround.

That's why I'd suggest to clean this up by using this module here for the `bw` lookup and to write another lookup to call `rbw` without needing a local bash wrapper. So people can decide for themselves which tool they want to use.

Would you agree this would be the cleanest solution to solve the bw-rbw issues?